src/common/librlist/rlist.c

```diff
@@ -1809,18 +1809,19 @@ static zlistx_t *rlist_get_nnodes (struct rlist *rl, int nnodes)
  *  Works by getting the first N least utilized nodes and spreading
  *  the nslots evenly across the result.
  */
-static struct rlist *rlist_alloc_nnodes (struct rlist *rl, int nnodes,
-                                         int cores_per_slot, int slots)
+static struct rlist *rlist_alloc_nnodes (struct rlist *rl,
+                                         struct rlist_alloc_info *ai)
```

Comment:
Nice cleanup!

Not sure if this is feasible, but in functions where the `struct rlist_alloc_info` parameter is not modified, can it be declared `const`?  Parameters that were obviously "in" before could now be sneakily "in/out", so the extra declaration could clarify that.

Reply:
Yeah, that sounds like a good suggestion!

Reply:
Ok, I've made that change and force-pushed the result. Thanks!